src/bin/scheduler/setup/api.rs

```diff
@@ -31,30 +38,33 @@ pub fn skip(plans: Vec<Plan>) -> (Box<dyn SetupStep>, Vec<Plan>) {
     (Box::new(SetupStepSuccess {}), plans)
 }
 
-pub fn run_steps(steps: Vec<StepWithPlans>) -> (Vec<Plan>, Vec<SetupFailure>) {
+pub fn run_steps(steps: Vec<StepWithPlans>) -> Result<(Vec<Plan>, Vec<SetupFailure>), Cancelled> {
     let mut plans = Vec::new();
     let mut errors = Vec::new();
     for (step, affected_plans) in steps.into_iter() {
         match step.setup() {
             Ok(()) => {
                 plans.extend(affected_plans);
             }
-            Err(err) => {
-                for plan in &plans {
-                    log::error!(
-                        "Plan {}: {}. Plan won't be scheduled.\nCaused by: {:?}",
-                        plan.id,
-                        err.summary,
-                        err.cause,
-                    );
-                    errors.push(SetupFailure {
-                        plan_id: plan.id.clone(),
-                        summary: err.summary.clone(),
-                        details: format!("{:?}", err.cause),
-                    });
+            Err(err) => match err {
+                Terminate::Cancelled => return Err(Cancelled),
```

Comment:
This feels like it will complicate all the setup steps. Would not be better idea to simply check after each setup step whether we should terminate? A setup step itself still can report whatever it want as an error.

Reply:
Done